csrc/preseg_passes/consecutive_cast.cpp

```diff
@@ -102,83 +299,61 @@ void castOptimizationPass(Fusion* fusion) {
     if (visited.count(expr) != 0 || !isCast(expr)) {
       continue;
     }
-    std::list<Val*> chain_cast_vals;
-    auto prev_expr = expr->input(0)->definition();
-    while (isCast(prev_expr)) {
-      auto intermediate_cast = prev_expr->output(0);
-      // 1.2 Note, if the output of prev_expr
-      //   is used by other operation(s); or
-      //   is a direct output from fusion
-      // we skip the casting chaining
-      if (intermediate_cast->isFusionOutput() ||
-          intermediate_cast->uses().size() > 1) {
-        break;
-      }
 
-      // adding prev_expr to visited node so we'll short-cut it.
-      visited.insert(prev_expr);
```

Comment:
I'm unsure we need or will still need `visited`.

Reply:
good call. I forgot we are traversing through exprs, instead of following the data flow.

Reply:
realized that I still need the visited here.

i.e. this function `removeChainedCasts(expr, visited)` could have removed previous exprs in the for loop, which means we cannot call `isCast(expr)` since those are dangling pointers now.
```
    if (visited.count(expr) != 0 || !isCast(expr)) {
      continue;
    }
```